executor/batch_point_get.go

```diff
@@ -350,6 +354,11 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 		handles = append(handles, e.handles[i])
 		if e.lock && rc {
 			existKeys = append(existKeys, key)
+			// when e.handles is set in builder directly, index should be primary key and the plan is CommonHandleRead
+			// with clustered index enabled, lock primary key for clustered index table is redundant
+			if i < len(indexKeys) {
```

Comment:
Could we use `if len(indexKeys) == 0` here for cluster index situation(or `!= 0` for the non-cluster situation)? It's difficult to maintain the relation betweeen index of `handleKeys` and the length of `indexKeys`.